gusto/eady_diagnostics.py

```diff
@@ -78,31 +53,19 @@ def compute(self, state):
         return self.field(state.mesh).interpolate(potential)
 
 
-class EadyTotalEnergy(DiagnosticField):
+class EadyTotalEnergy(Energy):
     name = "EadyTotalEnergy"
 
-    def field(self, mesh):
-        if hasattr(self, "_field"):
-            return self._field
-        self._field = Function(FunctionSpace(mesh, "DG", 0), name=self.name)
-        return self._field
-
     def compute(self, state):
         kinetic = KineticEnergy()
         potential = EadyPotentialEnergy()
         total = kinetic.compute(state) + potential.compute(state)
         return self.field(state.mesh).interpolate(total)
 
 
-class CompressibleEadyTotalEnergy(DiagnosticField):
+class CompressibleEadyTotalEnergy(Energy):
     name = "CompressibleEadyTotalEnergy"
 
```

Comment:
It feels like there should be a nice way to compute the total energy, given the kinetic and potential energies - we shouldn't need a separate class... let me think...